fault/tester/base.py

```diff
@@ -41,6 +46,24 @@ def __init__(self, circuit: m.Circuit, clock: m.Clock = None,
             raise TypeError(f"Expected reset port: {reset, type(reset)}")
         self.reset_port = reset
 
+    def _find_default_clock(self, ports):
+        """Automatically finds a clock if there is only one"""
+        clock = None
+        for port in ports:
+            next_clock = None
+            if isinstance(port, m.Clock):
+                next_clock = port
+            elif isinstance(port, (m.Array, m.Tuple)):
+                nested_clock = self._find_default_clock(port)
```

Comment:
does this need to be `self._find_default_clock([port])`

Reply:
Nope, because it will iterate over the port (over it's children), otherwise we would infinitely loop on this array port

Reply:
I changed it to use port.ts for clarity (this is what it was doing implicitly)

Reply:
got it now, thanks.